From: Rob Evers <revers@redhat.com>
Date: Wed, 17 Nov 2010 22:15:19 -0500
Subject: [scsi] lpfc: fix locking for security mailbox commands
Message-id: <1290032123-5584-5-git-send-email-revers@redhat.com>
Patchwork-id: 29482
O-Subject: [RHEL5.6 PATCH 4/8] lpfc: Fix locking issue for security mailbox
	commands.
Bugzilla: 649489

From: Rob Evers on behalf of Emulex <revers@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=649489

diff --git a/drivers/scsi/lpfc/lpfc_attr.c b/drivers/scsi/lpfc/lpfc_attr.c
index 385bac7..6ece7ec 100644
--- a/drivers/scsi/lpfc/lpfc_attr.c
+++ b/drivers/scsi/lpfc/lpfc_attr.c
@@ -4899,8 +4899,14 @@ sysfs_mbox_read(struct kobject *kobj, char *buf, loff_t off, size_t count)
 			break;
 		case MBX_SECURITY_MGMT:
 		case MBX_AUTH_PORT:
-			if (phba->pci_dev_grp == LPFC_PCI_DEV_OC)
+			if (phba->pci_dev_grp == LPFC_PCI_DEV_OC) {
+				printk(KERN_WARNING "mbox_read:Command 0x%x "
+				       "is not permitted\n", 
+				       sysfs_mbox->mbox->u.mb.mbxCommand);
+				sysfs_mbox_idle(phba, sysfs_mbox);
+				spin_unlock_irq(&phba->hbalock);
 				return -EPERM;
+			}
 			break;
 		case MBX_READ_SPARM64:
 		case MBX_READ_LA:
